### ISSUE TITLE
Undocumented/Non-standard binary time encoding prevents non-Golang Custom VMs from working

### ISSUE THREAD

**Issue by archis-polyverse** [ava-labs/avalanchego#1002]
**Describe the bug**
Avalanche uses `time.MarshalBinary` and `time.UnmarshalBinary` frequently. See this example:
https://github.com/ava-labs/avalanchego/blob/e388a890b6c02c0f58fd116357fe508e65bb1c42/vms/rpcchainvm/vm_client.go#L195

Why this is a problem: The time binary Marshaling is an undocumented golang-specific format, but worse yet, not part of the Go compatibility promise.
https://stackoverflow.com/questions/60529098/marshalbinary-time-time-predicting-its-byte-size

This means that some day when AvalancheGo nodes are compiled with different go versions, or plugins mismatch, it will inevitably fail.

TODAY this is a problem for anyone writing Avalanche custom VMs not using Go. 

**To Reproduce**
Use a non-Golang Custom VM. I am building one here: https://github.com/archisgore/landslide. I have readme steps to test it using [ava-sim](https://github.com/ava-labs/ava-sim).

You will notice these error in the ava-sim output:

```.bash
ERROR[02-04|14:35:32] chains/manager.go#269: error creating chain 29f5CvuHCtzc1MCBd8h88zFzggysFq76PiPy7iQ8QQF229bgSE: error while creating new snowman vm Time.UnmarshalBinary: unsupported version
ERROR[02-04|14:35:33] chains/manager.go#269: error creating chain 29f5CvuHCtzc1MCBd8h88zFzggysFq76PiPy7iQ8QQF229bgSE: error while creating new snowman vm Time.UnmarshalBinary: unsupported version
ERROR[02-04|14:35:33] chains/manager.go#269: error creating chain 29f5CvuHCtzc1MCBd8h88zFzggysFq76PiPy7iQ8QQF229bgSE: error while creating new snowman vm Time.UnmarshalBinary: unsupported version
ERROR[02-04|14:35:33] chains/manager.go#269: error creating chain 29f5CvuHCtzc1MCBd8h88zFzggysFq76PiPy7iQ8QQF229bgSE: error while creating new snowman vm Time.UnmarshalBinary: unsupported version
```

**Expected behavior**
This shouldn't error. There should be a documented language-independent marshaling format.

**Screenshots**
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/91631493/152613952-eda0d8ed-2877-4465-ac94-3fcbee91c460.png">

**Operating System**
MacOS Darwin

**Additional context**
I am happy to fix this problem and submit a PR. I'd appreciate someone telling me that this is of interest. Sooner or later someone else will be blocked by this as well.

If so, suggestions on what time format would be acceptable to the community.

To get the ball rolling I propose using UTC Unix timestamp in nanoseconds as a 128-bit little-endian integer.

By submitting this issue I agree to the [Terms and Conditions](https://files.avalabs.org/legal/bug_hunt_q2_2020_terms.pdf) of the [Developer Accelerator Program](https://www.avax.network/avalanche-x).
